src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -13,18 +13,26 @@ internal class AwsClient( IAmazonSecurityTokenService stsClient ) : IAwsClient {
 		AwsRole role,
 		int durationInMinutes
 	) {
-		var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
-			PrincipalArn = role.PrincipalArn,
-			RoleArn = role.RoleArn,
-			SAMLAssertion = samlAssertion,
-			DurationSeconds = durationInMinutes * 60,
-		} );
 
-		return new AwsCredentials(
-			SessionToken: authResp.Credentials.SessionToken,
-			AccessKeyId: authResp.Credentials.AccessKeyId,
-			SecretAccessKey: authResp.Credentials.SecretAccessKey,
-			Expiration: authResp.Credentials.Expiration.ToUniversalTime()
-		);
+		try {
+			var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
+				PrincipalArn = role.PrincipalArn,
+				RoleArn = role.RoleArn,
+				SAMLAssertion = samlAssertion,
+				DurationSeconds = durationInMinutes * 60,
+			} );
+
+			return new AwsCredentials(
+				SessionToken: authResp.Credentials.SessionToken,
+				AccessKeyId: authResp.Credentials.AccessKeyId,
+				SecretAccessKey: authResp.Credentials.SecretAccessKey,
+				Expiration: authResp.Credentials.Expiration.ToUniversalTime()
+			);
+		} catch( Exception ex ) {
+			if( ex.Message == "The requested DurationSeconds exceeds the MaxSessionDuration set for this role." ) {
```

Comment:
this case has a more specific exception type, right?
you can / should use a more specific `catch` clause in such scenarios, e.g.
```csharp
} catch ( AmazonSecurityTokenServiceException ex ) when ( ex.Message == "foo" ) {
	throw new BmxException( "don't do foo", ex );
} catch ( Exception ex ) {
	throw new BmxException( "don't do... idk... this", ex );
}
```